pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinResultSet.java

```diff
@@ -790,7 +793,7 @@ public String getString( int index ) throws SQLException {
 
   @Override
   public String getString( String columnName ) throws SQLException {
-    return getString( rowMeta.indexOfValue( columnName ) );
+    return getString( rowMeta.indexOfValue( columnName ) + 1 );
```

Comment:
If a column name isn't found on the row, getXXX( 0 ) will be called, which means currentRow[ -1 ] will be accessed
Another IndexOutOfBounds will be thrown. Is that okay?